python/pfs/drp/stella/fitPolynomial.py

```diff
@@ -0,0 +1,186 @@
+                break
+            good = newGood
+            numGood = newNumGood
+        else:
+            func = self.fit(xx, yy, good)
+            fit = self.characterize(xx, yy, good, func)
+            if self.debugInfo.plot:
+                self.plot(xx, yy, good, fit)
+        fit.good = good
+        return fit
+
+    def fit(self, xx, yy, good, xMin=None, xMax=None):
+        """Fit a polynomial
+
+        The rejection already determined is applied; no further rejection or
+        iteration is performed here.
+
+        Parameters
+        ----------
+        xx : array_like, shape ``(N,)``
+            Ordinate values.
+        yy : array_like, shape ``(N,)``
+            Coordinate values.
+        good : array_like of `bool`, shape ``(N,)``
+            Boolean array indicating whether each point should be used in the
+            fit.
+        xMin, xMax : `int`, optional
+            Minimum and maximum ordinate values.
+
+        Returns
+        -------
+        func : callable
+            Function that returns the fit value.
+        """
+        domain = (xMin, xMax) if xMin is not None and xMax is not None else None
+        return np.polynomial.Chebyshev.fit(xx[good], yy[good], self.config.order, domain=domain)
+
+    def characterize(self, xx, yy, good, func):
+        """Characterize the polynomial fit
+
+        Parameters
+        ----------
+        xx : array_like, shape ``(N,)``
+            Ordinate values.
+        yy : array_like, shape ``(N,)``
+            Coordinate values.
+        good : array_like of `bool`, shape ``(N,)``
+            Boolean array indicating whether each point should be used in the
+            fit.
+        func : callable
+            Function that returns the fit value.
+
+        Returns
+        -------
+        func : callable
+            Function that returns the fit value.
+        fit : `numpy.ndarray`, shape ``(N,)``
+            Values of the function at each of the input ``xx`` values.
+        residuals : `numpy.ndarray`, shape ``(N,)``
+            Fit residuals (``yy - fit``).
+        rms : `float`
+            Robust RMS of the residuals (from quartiles).
+        """
+        fit = func(xx)
+        residuals = yy - fit
+        lq, uq = np.percentile(residuals[good], [25.0, 75.0])
+        rms = 0.741*(uq - lq)
+        return Struct(func=func, fit=fit, residuals=residuals, rms=rms)
+
+    def plot(self, xx, yy, good, fit):
+        """Plot the polynomial fit
+
+        Parameters
+        ----------
+        xx : array_like, shape ``(N,)``
+            Ordinate values.
+        yy : array_like, shape ``(N,)``
+            Coordinate values.
+        good : array_like of `bool`, shape ``(N,)``
+            Boolean array indicating whether each point should be used in the
+            fit.
+        fit : `lsst.pipe.base.Struct`
+            Fit results from the ``fit`` method.
+        """
+        import matplotlib.pyplot as plt
```

Comment:
`matplotlib.pyplot` can be a troublesome import in a library, because it sets some globals that can't be unset by the user. For this reason, it's often a good idea to only import it when you need it.